### ISSUE TITLE
Feature Request: Access from lan/localhost/web

### ISSUE THREAD

**Issue by Gibbz** [tbar0970/jethro-pmm#93]
Currently in the config file is a place to setup the domain name. Currently this is the only way to access the database via this address.
It would be real handy not to have to put this in, so that the database can be accessed from multiple address'.

For example id like to be able to use
localhost if on the machine with no network
the lan ip if there is no internet
the web/domain name is useful for accessing outside the lan

**Comment by tbar0970** [comment]
As a workaround, you could set up several copies of the code, with different BASE_URLs, pointing to the same database.

**Comment by Gibbz** [comment]
Do you mean setup multiple websites? Or is there a way to have multiple configurations?

**Comment by tbar0970** [comment]
Yes, set up multiple installations of Jethro - eg localhost/jethro1, mysite.domain.com/jethro2 etc - but use the same database DSN in each one.

**Comment by jefft** [comment]
+1 for eliminating BASE_URL and just inferring the URL from the request.

We're running Jethro on a server in the church office. In addition to the external URL, we'd like to be able to access Jethro from an internal URL (http://jethro) when the internet goes down. We currently achieve this with two copies of Jethro, but I'm sure this isn't a good idea.

**Comment by Gibbz** [comment]
That would be great!
If anyone wants to force a redirect anyway, they can use the .htaccess file
far as i know...

**Comment by Gibbz** [comment]
Right, i found the code for this in init.php
Ive just put a block comment around this using  /* and  */ so this disables this annoying feature!
Hope this is useful to other people :)

/*
if (php_sapi_name() != 'cli') {
	// Make sure we're at the correct URL
	$do_redirect = FALSE;
	if (REQUIRE_HTTPS && !defined('IS_PUBLIC') && empty($_SERVER['HTTPS'])) {
		$do_redirect = TRUE;
	}
	if (strpos(array_get($_SERVER, 'HTTP_HOST', array_get($_SERVER, 'SERVER_NAME', '')).$_SERVER['REQUEST_URI'], str_replace(Array('http://', 'https://'), '', BASE_URL)) !== 0) {
		$do_redirect = TRUE;
	}
	if ($do_redirect) {
		header('Location: '.build_url(Array()));
		exit();
	}
}
*/

**Comment by Gibbz** [comment]
Note: this does cause some wierd issues with Jethro :(

ive noticed if you click on families or people it works for the first page, but then hold the mouse over the other pages and the url is changed from domainname.com to the BASEURL which is a bit weird.... So it doesnt work

**Comment by Gibbz** [comment]
Any news on this request at all?

**Comment by vanoudt** [comment]
Might I suggest that Jethro is not the best place for this. If you simply use your computers `hosts` functionality, you could set the internet domain to resolve to your local machine. That way, when the internet goes down, you'd notice no difference.
See [here](http://www.makeuseof.com/tag/modify-manage-hosts-file-linux/) for linux instructions, or [here](http://www.howtogeek.com/howto/27350/beginner-geek-how-to-edit-your-hosts-file/) for windows.

**Comment by tbar0970** [comment]
I don't expect I'll have time to do this as unsponsored work any time soon, to be honest.

If you'd like to talk about booking it in as sponsored dev, let me know.  You can get me via my github username @gmail.
